lms/djangoapps/completion/serializers.py

```diff
@@ -0,0 +1,212 @@
+
+    Usage:
+
+    To create CourseAggregationAdapters for a user's courses with a given queryset:
+
+        >>> from lms.djangoapps.completion.models import AggregateCompletion
+        >>> from lms.djangoapps.completion.serializers import CourseAggregationAdapter
+        >>> completions = AggregateCompletion.objects.filter(
+        >>>     user=user,
+        >>>     aggregation_name__in=['course', 'chapter', 'vertical']
+        >>> )
+        >>> adapters = []
+        >>> for course_key in Enrollments.objects.filter(user=user, active=True):
+        >>>     adapters.append(CourseAggregationAdapter(
+        >>>         user=user,
+        >>>         course_key=course_key,
+        >>>         queryset=completions,
+        >>>     ))
+
+    To add an aggregation or iterable of aggregations to an adapter:
+
+        >>> from lms.djangoapps.completion.serializers import CourseAggregationAdapter
+        >>> adapter = CourseAggregationAdapter(
+        >>>     user=user,
+        >>>     course_key=course_key,
+        >>> )
+        >>> adapter.add_aggregate_completion(completion1)
+        >>> adapter.update_aggregate_completions([completion2, completion3])
+
+    The adapter or list of adapters can then be passed to the serializer for processing.
+    """
+    def __init__(self, user, course_key, queryset=None):
+        self.user = user
+        self.course_key = course_key
+        self.aggregations = defaultdict(list)
+        if queryset:
+            self.update_aggregate_completions(queryset)
+
+    def __getattr__(self, name):
+        """
+        Provide the serializer with access to custom aggregators
+        """
+        if is_aggregation_name(name):
+            return self.aggregations.get(name, [])
+        else:
+            raise AttributeError
+
+    def add_aggregate_completion(self, aggregation):
+        """
+        Add an aggregation to the CourseAggregationAdapter, checking
+        whether it meets the criteria for user, course_key, and aggregation_name
+        """
+        if (aggregation.user, aggregation.course_key) == (self.user, self.course_key):
+            if is_aggregation_name(aggregation.aggregation_name):
+                self.aggregations[aggregation.aggregation_name].append(aggregation)
+
+    def update_aggregate_completions(self, iterable):
+        for aggregation in iterable:
+            self.add_aggregate_completion(aggregation)
+
+    @property
+    def course(self):
+        """
+        The AggregateCompletion for the course as a whole.
+
+        If no course completion exists, use a dummy completion
+        """
+        if self.aggregations['course']:
+            return self.aggregations['course'][0]
+        else:
+            return AggregateCompletion(
+                user=self.user,
+                course_key=self.course_key,
+                aggregation_name='course',
+                earned=0.0,
+                possible=None,
```

Comment:
In the views you made "create_dummy_aggregation" where the field 'possible' was 0.0 any reason why this is different? What about setting percent to 0.0 here and in create_dummy_aggregation as well?